projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,234 @@
+        lawsuit = lawsuit.replace("&lt;companyAddress&gt;", companyAddress);
+        lawsuit = lawsuit.replace("&lt;companyPhone&gt;", companyPhone);
+        lawsuit = lawsuit.replace("&lt;companyFax&gt;", companyFax);
+
+        // Second spam company
+        lawsuit = lawsuit.replace("&lt;company2Name&gt;", company2Name);
+        lawsuit = lawsuit.replace("&lt;company2Id&gt;", company2Id);
+        lawsuit = lawsuit.replace("&lt;company2Address&gt;", company2Address);
+        lawsuit = lawsuit.replace("&lt;company2Phone&gt;", company2Phone);
+        lawsuit = lawsuit.replace("&lt;company2Fax&gt;", company2Fax);
+
+        return lawsuit;
+    }
+
+    private String readTemplate() {
+
+        StringBuilder contentBuilder = new StringBuilder();
+        try {
+            BufferedReader in = new BufferedReader(new FileReader(lawsuitTemplateFilePath));
+            String str;
+            while ((str = in.readLine()) != null) {
+                contentBuilder.append(str);
+            }
+            in.close();
+        } catch (IOException e) {
+            Log.w(TAG, "Error reading lawsuit's template.");
+            e.printStackTrace();
+        }
+        return contentBuilder.toString();
+    }
+
+    private String getPdfFilename() {
+
+        return dateAndTimeFormatter.format(new Date()) + ".pdf";
+    }
+
+    private void initDirStructure() {
+
+        lawsuitMainPath = Paths.get(Environment.getExternalStorageDirectory().getPath(), getString(R.string.app_name), LAWSUIT_MAIN_DIR_NAME).toString();
+        initDir(lawsuitMainPath);
+
+        lawsuitTemplateDirPath = Paths.get(lawsuitMainPath, LAWSUIT_TEMPLATE_DIR_NAME).toString();
+        initDir(lawsuitTemplateDirPath);
+
+        // TODO: Currently template's file is been stored in user's device storage. Check to see if this should be part of the apk package.
+        try {
+            lawsuitTemplateFilePath = Paths.get(lawsuitTemplateDirPath, LAWSUIT_TEMPLATE_FILE_NAME).toString();
+        } catch (Exception e) {
+            Toast.makeText(this, "No template file found for the lawsuit.",
+                    Toast.LENGTH_LONG).show();
+        }
+
+        // Output
+        lawsuitOutputPath = Paths.get(lawsuitMainPath, LAWSUIT_OUTPUT_DIR_NAME).toString();
+        initDir(lawsuitOutputPath);
+    }
+
+    /**
+     * Create dir if it doesn't exists.
+     * @param path Directory to look for.
+     */
+    private void initDir(String path) {
+
+        File directory = new File(lawsuitTemplateDirPath);
+        if (directory.exists() && directory.isDirectory())
+            directory.mkdirs();
+    }
+
+    private void getWriteStoragePermission() {
+
+        if (Build.VERSION.SDK_INT >= 23)
+            if (!(checkSelfPermission(android.Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED))
+                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 2);
```

Comment:
What happens if the user doesn't grant the permissions?

Reply:
Changed it to be onClick as follows:
1. First time - request permission.
2. Denies - returns back to form.
3. Clicks again - Explain in a dialog why permission is need, and then asks for permission again.
    This will be continuous on every click attempt, unless user has checked "Do not show this 
    again" .
4. Click after "Do not show this again" - Only shows toast "Write storage permission needed."